src/core_plugins/kibana/public/home/components/home.js

```diff
@@ -38,9 +38,61 @@ import {
 
 import { FeatureCatalogueCategory } from 'ui/registry/feature_catalogue';
 
-export function Home({ addBasePath, directories, apmUiEnabled, recentlyAccessed }) {
+export class Home extends Component {
 
-  const renderDirectories = (category) => {
+  state = {
+    isLoading: false,
+    isNewKibanaInstance: false,
+  }
+
+  componentWillUnmount() {
+    this._isMounted = false;
+  }
+
+  componentDidMount() {
+    this._isMounted = true;
+    this.fetchIsNewKibanaInstance();
+  }
+
+  fetchIsNewKibanaInstance = async () => {
+    this.setState({
```

Comment:
I think this is a case where promises are a good bit cleaner. I think this entire block could be rewritten like this, and  would be easier to grok:

```js
this.setState({ isLoading: true });

this.props.find({
  type: 'index-pattern',
  fields: ['title'],
  search: `*`,
  search_fields: ['title'],
  perPage: 1
})
  .catch(() => ({
    // Ignore errors, as this is just a cosmetic detail
    total: 0
  }))
  .then(({ total }) => this._isMounted && this.setState({ isLoading: false, isNewKibanaInstance: total === 0 }))
```